# HG changeset patch
# User Luca Greco <lgreco@mozilla.com>
# Date 1457855811 -3600
#      Sun Mar 13 08:56:51 2016 +0100
# Node ID 906e3798d90629405dc35cfe70b8f98e6f265a8f
# Parent  3bc413216cba4990c1d98eb07bc4eee133d7d419
FIXUP: revert changes on content/worker destroy

MozReview-Commit-ID: 8Gi7y7851eD

diff --git a/addon-sdk/source/lib/sdk/content/worker.js b/addon-sdk/source/lib/sdk/content/worker.js
--- a/addon-sdk/source/lib/sdk/content/worker.js
+++ b/addon-sdk/source/lib/sdk/content/worker.js
@@ -158,22 +158,23 @@ connect.define(Worker, function(worker, 
 detach.define(Worker, function(worker) {
   let model = modelFor(worker);
   if (!model.attached)
     return;
 
   processes.port.off('sdk/worker/event', worker.receive);
   model.attached = false;
   model.destroyed = true;
+  emit(worker, 'detach');
 });
 
 isPrivate.define(Worker, ({ tab }) => isPrivate(tab));
 
 // Something in the parent side has destroyed the worker, tell the child to
 // detach, the child will respond when it has detached
 destroy.define(Worker, function(worker, reason) {
   let model = modelFor(worker);
-  if (!model.attached || model.destroyed)
+  model.destroyed = true;
+  if (!model.attached)
     return;
 
-  model.destroyed = true;
   worker.send('detach', reason);
 });
